File: Src/Icm.Core/Localization/ResourceLocalizationRepository.vb
Namespace Icm.Localization

    Public Class ResourceLocalizationRepository
        Implements ILocalizationRepository

        Private ReadOnly _resourceManager As System.Resources.ResourceManager

        Public Sub New(rman As System.Resources.ResourceManager)
            _resourceManager = rman
        End Sub

        Public ReadOnly Property ItemForCulture(lcid As Integer, key As String) As String Implements ILocalizationRepository.ItemForCulture
            Get
                Dim culture = CultureInfo.GetCultureInfo(lcid)
                Return If(_resourceManager.GetString(key, culture), key)
            End Get
        End Property
    End Class

End Namespace

File: Src/Icm.Core/Localization/ResourceLocalizationRepository.cs
using System.Globalization;

namespace Icm.Localization
{

	public class ResourceLocalizationRepository : ILocalizationRepository
	{


		private readonly System.Resources.ResourceManager _resourceManager;
		public ResourceLocalizationRepository(System.Resources.ResourceManager rman)
		{
			_resourceManager = rman;
		}

	    public string this[int lcid, string key]
	    {
	        get {
                var culture = CultureInfo.GetCultureInfo(lcid);
                return _resourceManager.GetString(key, culture) ?? key;
            }
	    }
	}

}
